gulpfile.js

```diff
@@ -0,0 +1,110 @@
+var gulp = require('gulp')
+var Server = require('karma').Server
+var mocha = require('gulp-mocha')
+var ipfsd = require('ipfsd-ctl')
+var fs = require('fs')
+
+gulp.task('default', function () {
+  gulp.start('test:node')
+})
+
+gulp.task('test:node', function (done) {
+  startDisposableDaemons(function (daemons) {
+    gulp.src('test/tests.js')
+      // gulp-mocha needs filepaths so you can't have any plugins before it
+      .pipe(mocha())
+      .once('error', function () {
+        process.exit(1)
+      })
+      .once('end', function () {
+        stopDisposableDaemons(daemons, function () {
+          process.exit()
```

Comment:
However, now I have this problem, the solution recommended by gulp-mocha is to do process.exit() in case there is some callback on the eventloop waiting for its event (which happens), which makes the process exit without finishing the browser tests. Any elegant way you would recommend to do this @Dignifiedquire ? Thank you :)